tracer/build/_build/Build.Shared.Steps.cs

```diff
@@ -138,15 +151,20 @@ partial class Build
         .After(CompileNativeLoader)
         .Executes(() =>
         {
-            foreach (var (arch, folder, extension) in GetMacOsArchitectureAndExtension())
-            {
-                var source = NativeLoaderProject.Directory / "bin" / "loader.conf";
-                var dest = MonitoringHomeDirectory / folder;
-                CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
-                    
-                CopyFile(NativeLoaderProject.Directory / "bin" / $"{NativeLoaderProject.Name}.{arch}.{extension}",
-                         MonitoringHomeDirectory / folder / $"{NativeLoaderProject.Name}.{extension}",
-                         FileExistsPolicy.Overwrite);
-            }
+            // Copy loader.conf
+            var source = NativeLoaderProject.Directory / "bin" / "loader.conf";
+            var dest = MonitoringHomeDirectory / "osx";
+            CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+            
+            // Create universal shared library with all archs in a single file
+            var lstNativeBinaries = OsxArchs
+               .Select(arch => NativeLoaderProject.Directory / "bin" / $"{NativeLoaderProject.Name}.{arch}.dylib");
+
+            var destination = $"./{NativeLoaderProject.Name}.dylib";
+            var destFolder = MonitoringHomeDirectory / "osx";
+            Console.WriteLine($"Creating universal binary for {destination}");
+            EnsureExistingDirectory(destFolder);
+            var strNativeBinaries = string.Join(' ', lstNativeBinaries);
+            Lipo.Value(arguments: $"{strNativeBinaries} -create -output {destFolder}/{destination}");
```

Comment:
Same comment here, could we move the generation of the universal binary into the compile step so the publish step is only concerned with copying to the final destination?

Reply:
done in [62fda46](https://github.com/DataDog/dd-trace-dotnet/pull/3454/commits/62fda46102ad930801ed65d514e8ed655da5dea3)